src/main/scala/is/hail/variant/VariantSampleMatrix.scala

```diff
@@ -224,8 +224,9 @@ object VSMSubgen {
 }
 
 class VariantSampleMatrix[RPK, RK, T >: Null](val hc: HailContext, val metadata: VSMMetadata,
```

Comment:
We store the `VSMMetadata` twice, in VSM and in `MatrixValue`. Can we define VSM's metadata in terms of its `matrixType` field?